### ISSUE TITLE
"Nonce Mismatch" for parallel acquireTokenSilent() calls

### ISSUE THREAD

**Issue by bwiesmueller** [AzureAD/microsoft-authentication-library-for-js#1132]
## I'm submitting a...

<pre><code>
[ ] Regression (a behavior that used to work and stopped working in a new release)
[X] Bug report  
[ ] Performance issue
[ ] Feature request
[ ] Documentation issue or request
[ ] Other... Please describe:
</code></pre>

## Browser:
- [X] Chrome version 78.0.3904.108
- [ ] Firefox version XX
- [ ] IE version XX
- [ ] Edge version XX
- [ ] Safari version XX

## Library version

<pre><code>
Library version: 1.1.3

</code></pre>

## Current behavior

When the user is already signed in without Msal (here via Azure App Service EasyAuth) and sending multiple non-cached acquireTokenSilent() calls with a login hint for different scopes in parallel, the operation fails with a "Nonce Mismatch" error.

This only happens when the login response contains an idToken, e.g. when using a login hint in the token request (see the example below).
https://github.com/AzureAD/microsoft-authentication-library-for-js/blob/9c02f2604bd9fa2d2e4de777ffdc28e977091cdf/lib/msal-core/src/UserAgentApplication.ts#L1682

Both requests will set the "msal.nonce.idtoken" variable in local storage.
https://github.com/AzureAD/microsoft-authentication-library-for-js/blob/9c02f2604bd9fa2d2e4de777ffdc28e977091cdf/lib/msal-core/src/UserAgentApplication.ts#L2246
The second request overwrites the value of the first one. When processing the response of the first request a nonce mismatch is reported (as it will read the nonce value from the second request from local storage).
https://github.com/AzureAD/microsoft-authentication-library-for-js/blob/9c02f2604bd9fa2d2e4de777ffdc28e977091cdf/lib/msal-core/src/UserAgentApplication.ts#L1710

As a workaround I am now serializing all my acquireTokenSilent() calls.

## Expected behavior

Sending such requests in parallel should be possible without an error.

## Minimal reproduction of the problem with instructions

`index.html` (removed scopes, and application-specific IDs)

```<!DOCTYPE html>
<html>
  <head>
    <title>Request access tokens</title>
    <script src="msal.js"></script>
  </head>
  <body>
    <script>
      localStorage.clear(); // Problem does not occur when using cached access tokens

      const msalConfig = {
        auth: {
          clientId: '<clientId>',
          authority: '<authority>',
          redirectUri: 'http://localhost:4200/assets/login-response.html'
        },
        cache: {
          cacheLocation: 'localStorage',
          storeAuthStateInCookie: false
        }
      };

      const msal = new Msal.UserAgentApplication(msalConfig);

      msal.acquireTokenSilent({
        scopes: ['<scope1>'],
        loginHint: '<usermail>'
      }).then(tokenResponse => {
        console.log("Token response 1: ", tokenResponse);
      }).catch(error => {
        console.error("ERROR 1: ", error); // Will fail with error "ClientAuthError: Nonce is not matching, Nonce received:  <nonce1>, nonce expected :<nonce>2".
      });

      msal.acquireTokenSilent({
        scopes: ['<scope2>'],
        loginHint: '<usermail>'
      }).then(tokenResponse => {
        console.log("Token response 2: ", tokenResponse); // Will succeed
      }).catch(error => {
        console.error("ERROR 2: ", error);
      });
    </script>
  </body>
</html>
```
The user is already logged in here, I am using loginHint as described here: https://docs.microsoft.com/en-us/azure/active-directory/develop/msal-js-sso#sso-without-msaljs-login

**Comment by jasonnutter** [comment]
@bwiesmueller Can you please try `1.2.0-beta.4` and let us know if you still see the issue? I believe we fixed this bug in the latest betas. @sameerag can confirm.

**Comment by sameerag** [comment]
Yes, this should be fixed in `1.2.0-beta.4`. Please test it and let us know?

**Comment by bwiesmueller** [comment]
Yes, you are right. It is fixed in `1.2.0-beta.4` and works for my example :)

**Comment by sameerag** [comment]
Great! Thanks for the feedback. Closing this issue.
